app/javascript/components/MapAndButtons.jsx

```diff
@@ -146,7 +146,7 @@ class MapAndButtons extends React.Component {
   toggleEditHandler = (e) => this.setState({ edit_mode: !this.state.edit_mode })
 
   build_per_floor_icon_set = (locations) => {
-    return locations.filter(l => l.floor_id === this.state.current_selected_floor).map(loc => Object.assign({}, {icon_url: loc.icon_url, icon_name: loc.icon_name}))
+    return locations.filter(l => l.floor_id === this.state.current_selected_floor).filter(l => l.icon_name != null).map(loc => Object.assign({}, {icon_url: loc.icon_url, icon_name: loc.icon_name}))
```

Comment:
combine both `.filter` into one with `&&` evalution.. Please check if `l.icon_name` is null or undefined.

Reply:
:+1: